subsys/bluetooth/services/fast_pair/adv_manager/core.c

```diff
@@ -0,0 +1,730 @@
+	 */
+	__ASSERT_NO_MSG(!k_is_preempt_thread());
+	__ASSERT_NO_MSG(!k_is_in_isr());
+
+	__ASSERT_NO_MSG(is_enabled);
+
+	LOG_INF("Fast Pair Adv Manager: RPA expired");
+
+	if (!uptime) {
+		uptime = k_uptime_get();
+	} else {
+		LOG_INF("Fast Pair Adv Manager: the last timeout has occurred %" PRId64 " [s] ago",
+			(k_uptime_delta(&uptime) / MSEC_PER_SEC));
+	}
+
+	/* In the FMDN provisioned state, the FMDN advertising set is responsible
+	 * for setting the global RPA timeout using the bt_le_set_rpa_timeout API.
+	 * In the unprovisioned state, it is the responsibility of the Fast Pair advertising set.
+	 */
+	if (!fmdn_provisioned) {
+		next_rpa_timeout = fp_adv_rpa_timeout_calculate();
+		err = bt_le_set_rpa_timeout(next_rpa_timeout);
+		if (err && (err != -EALREADY)) {
+			LOG_ERR("Fast Pair Adv Manager: "
+				"bt_le_set_rpa_timeout failed: %d for %d [s]",
+				err, next_rpa_timeout);
+		} else {
+			LOG_INF("Fast Pair Adv Manager: setting RPA timeout to %d [s]",
+				next_rpa_timeout);
+		}
+	}
+
+	if (fp_adv_manager_is_rpa_suspended()) {
+		LOG_INF("Fast Pair Adv Manager: RPA rotation is in the suspended mode");
+
+		expire_rpa = false;
+	}
+
+	if (bt_fast_pair_adv_manager_is_adv_active()) {
+		/* Update the advertising payload. */
+		err = fp_adv_payload_set(expire_rpa, false);
+		if (err) {
+			LOG_ERR("Fast Pair Adv Manager: "
+				"cannot set advertising payload (err: %d)", err);
+		}
+	}
+
+	return expire_rpa;
+}
+
+static int fp_adv_set_setup(void)
+{
+	int err;
+	static const struct bt_le_ext_adv_cb fp_adv_set_cb = {
+		.connected = fp_adv_connected,
+		.rpa_expired = fp_adv_rpa_expired,
+	};
+
+	__ASSERT(!fp_adv_set,
+		 "Fast Pair Adv Manager: invalid state of the advertising set");
+
+	/* Create the Fast Pair advertising set. */
+	err = bt_le_ext_adv_create(&fp_adv_param, &fp_adv_set_cb, &fp_adv_set);
+	if (err) {
+		LOG_ERR("Fast Pair Adv Manager: bt_le_ext_adv_create returned error: %d", err);
+		return err;
+	}
+
+	LOG_INF("Fast Pair Adv Manager: prepared the advertising set");
+
+	return 0;
+}
+
+static int fp_adv_set_teardown(void)
+{
+	int err;
+
+	__ASSERT(fp_adv_set, "Fast Pair Adv Manager: invalid state of the advertising set");
+
+	err = bt_le_ext_adv_delete(fp_adv_set);
+	if (err) {
+		LOG_ERR("Fast Pair Adv Manager: bt_le_ext_adv_delete returned error: %d", err);
+		return err;
+	}
+
+	fp_adv_set = NULL;
+
+	return 0;
+}
+
+static bool fp_adv_is_any_request_enabled(void)
+{
+	return (fp_adv_request_bm != 0);
+}
+
+static void fp_adv_restart_work_handle(struct k_work *w)
+{
+	if (bt_fast_pair_adv_manager_is_ready() && fp_adv_is_any_request_enabled()) {
+		fp_advertising_state_update(true);
+	}
+}
+
+static void fp_adv_conn_clear(void)
+{
+	int err;
+
+	if (fp_conn) {
+		err = bt_conn_disconnect(fp_conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+		if (err) {
+			LOG_ERR("Fast Pair Adv Manager: bt_conn_disconnect returned error: %d",
+				err);
+		}
+	}
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
```

Comment:
Log-only connection and disconnection that are associated with the Fast Pair advertising set